src/WiFi101.h

```diff
@@ -144,10 +146,9 @@ class WiFiClass
 	int hostByName(const char* hostname, IPAddress& result);
 	int hostByName(const String &hostname, IPAddress& result) { return hostByName(hostname.c_str(), result); }
 
-	uint8_t ping(const char* hostname, uint8_t ttl = 128);
-	uint8_t ping(const String &hostname, uint8_t ttl = 128);
-	uint8_t ping(IPAddress host, uint8_t ttl = 128);
-	uint8_t pingGateway();
+	wl_ping_result_t ping(const char* hostname, uint8_t ttl = 128);
```

Comment:
I prefer returning an enum type here, however it's inconsistent with the other WiFi methods, as they use `uint8_t` to return enum's. For, example both `status()` and `begin()` use return type `uint8_t` but value will be `wl_status_t`. The original [WiFi](https://github.com/arduino-libraries/WiFi) library also used this style.

@cmaglie @facchinm any thoughts on this?